mergin/test/test_client.py

```diff
@@ -2161,6 +2161,54 @@ def test_project_metadata(mc):
     assert mp.version() == "v0"
 
 
+def test_project_rename(mc: MerginClient):
+    """Check project can be renamed"""
+
+    test_project = "test_project_rename"
+    test_project_renamed = "test_project_renamed"
+    project = API_USER + "/" + test_project
+    project_renamed = API_USER + "/" + test_project_renamed
+
+    project_dir = os.path.join(TMP_DIR, test_project)  # primary project dir
+
+    cleanup(mc, project, [project_dir])
+    cleanup(mc, project_renamed, [])
+
+    shutil.copytree(TEST_DATA_DIR, project_dir)
+    mc.create_project_and_push(project, project_dir)
+
+    # renamed project does not exist
+    with pytest.raises(ClientError, match="The requested URL was not found on the server"):
+        info = mc.project_info(project_renamed)
+
+    # rename
+    mc.rename_project(project, test_project_renamed)
+
+    # validate project info
+    project_info = mc.project_info(project_renamed)
+    assert project_info["version"] == "v1"
+    assert project_info["name"] == test_project_renamed
+    assert project_info["namespace"] == API_USER
+
```

Comment:
also worth checking that the original project name does not exist anymore (project_info() will throw client error)